projects/pastanaga/src/lib/controls/filtered-checkbox-group/directory.component.ts

```diff
@@ -0,0 +1,80 @@
+import { Component, ChangeDetectionStrategy, Input, Output, EventEmitter, ChangeDetectorRef } from '@angular/core';
+import { markForCheck } from '../../common/utils';
+import { ControlModel } from '../control.model';
+
+const firstLetterRegExp = new RegExp('[a-zA-Z]');
+
+@Component({
+    selector: 'pa-directory',
+    templateUrl: 'directory.component.html',
+    styleUrls: ['./directory.component.scss'],
+    changeDetection: ChangeDetectionStrategy.OnPush,
+})
+
+export class DirectoryComponent {
+    @Input() set values(values: ControlModel[]) {
+        this._values = values;
+        this.createMap();
+    }
+    directory = ['#', 'A', 'B', 'C', 'D', 'E', 'F', 'G', 'H', 'I', 'J', 'K', 'L', 'M', 'N', 'O', 'P', 'Q', 'R', 'S', 'T', 'U',
+                 'V', 'W', 'X', 'Y', 'Z'];
+    labelMap?: { [key: string]: string[] };
+    _selectedLetter = '';
+    _values: ControlModel[] = [];
+    @Output() selected: EventEmitter<{letter: string, selection: string[]}> = new EventEmitter();
+
+    constructor(private cdr: ChangeDetectorRef) {}
+
+    createMap() {
+        const map: {[key: string]: any[]} = {'#': []};
```

Comment:
why do you initialize `#` manually while you have it in `directory`?

Also: is `any` really needed? couldn't you have a better typing? Reading the code following, I feel like you could have `string[]` instead

Reply:
fixed